### PR TITLE
Promisified everything, passing all tests

### PR DESCRIPTION
**Pull request by erjoalgo**

### CODE REVIEW

File: lib/async-storage.js [hunk 1]
```diff
@@ -1,274 +1,271 @@
...
-	    e.target.transaction.onerror = database.onerror;
-
-	    if(db.objectStoreNames.contains("items")) {
-		db.deleteObjectStore("items");
-	    }
-	    var store = db.createObjectStore("items");
-	};
+    if (opened  != null)	{
+	// throw new Error ("must close before opening again");
+    }else 	{
```
Comment (canuckistani) [inline lib/async-storage.js]: nit: please follow the code formatting conventions of the file.

File: lib/async-storage.js [hunk 2]
```diff
@@ -1,274 +1,271 @@
...
+    });
 }
 
-function _getItems(callback) {
-  let db = database.db;
-  let trans = db.transaction(["items"], "readwrite");
-  let store = trans.objectStore("items");
-  let items = [];
+function _getItems() {
+    return new Promise(function(resolve, reject){
```
Comment (canuckistani) [inline lib/async-storage.js]: nite: space between ) and {

File: lib/async-storage.js [hunk 3]
```diff
@@ -1,274 +1,271 @@
...
-    if (error === 'QuotaExceededError') {
-      callback(error);
-    }
-  };
+	cursorRequest.onerror = reject;
+    });
 }
 
+function _removeItem(key) {
+    return new Promise(function(resolve, reject){
```
Comment (canuckistani) [inline lib/async-storage.js]: same as above

File: lib/async-storage.js [hunk 4]
```diff
@@ -1,274 +1,271 @@
...
-      } else {
-        // When we get here, we've got the nth key.
-        if (callback) {
-          callback(null, cursor.key);
-        }
-      }
-    }
-  };
+function _clear() {
+    return new Promise(function(resolve, reject){
```
Comment (canuckistani) [inline lib/async-storage.js]: same as above

File: lib/async-storage.js [hunk 5]
```diff
@@ -1,274 +1,271 @@
...
+    });
 }
 
-function _keys(callback) {
-  let db = database.db;
-  let store = db.transaction(["items"], 'readonly').objectStore("items");
-  let req = store.openCursor();
-  let _keys = [];
+function _key(n) {
+    return new Promise(function(resolve, reject){
```
Comment (canuckistani) [inline lib/async-storage.js]: same as above

File: lib/async-storage.js [hunk 6]
```diff
@@ -1,274 +1,271 @@
...
+}
 
-    if (!cursor) {
-      if (callback) {
-        callback(null, _keys);
-      }
-      return;
-    }
+function _keys() {
+    return new Promise(function(resolve, reject){
```
Comment (canuckistani) [inline lib/async-storage.js]: same as above

File: lib/async-storage.js [hunk 7]
```diff
@@ -1,274 +1,271 @@
...
+
+UnopenedAsyncDB.prototype = new Error(); 
+UnopenedAsyncDB.prototype.constructor = UnopenedAsyncDB;
+
 function openPromisify ( func )	{
     return function(){
 	if (opened == null)	{
-	    throw new Error("must call open before any other async-storage method");
+	    throw new UnopenedAsyncDB(func.name);
 	}else 	{
```
Comment (canuckistani) [inline lib/async-storage.js]: nite: same formatting issue as before.

**Comment by canuckistani** [comment]
This looks really good, thanks for stepping in. I've got a few formatting nits, just to follow who the file is currently formatted. It's also a reminder to me to implement eslint checks for my projects, I'll try to get to that later today.

**Comment by erjoalgo** [comment]
I added another commit using jsfmt https://github.com/rdio/jsfmt. (similar to gofmt in case you're familiar with that). Hopefully it looks better more faithful now to the original convention
And thank you. It's nice to be able to abstract away from all the IndexedDB details.

**Comment by jeffgca** [comment]
Looks great, probably cleaner than my original code :)

**Pull request merged**